bundles/remote_services/discovery_shm/src/discovery_shmWatcher.c

```diff
@@ -165,20 +167,15 @@ static void* discoveryShmWatcher_run(void* data) {
     }
 
     if (endpointDiscoveryServer_getUrl(discovery->server, &url[0], MAX_LOCALNODE_LENGTH) != CELIX_SUCCESS) {
-        snprintf(url, MAX_LOCALNODE_LENGTH, "http://%s:%s/%s", DEFAULT_SERVER_IP, DEFAULT_SERVER_PORT, DEFAULT_SERVER_PATH);
+        snprintf(url, MAX_LOCALNODE_LENGTH, "http://%s:%s/%s", DEFAULT_SERVER_IP, DEFAULT_SERVER_PORT,
+                 DEFAULT_SERVER_PATH);
     }
 
-    while (watcher->running) {
-        // register own framework
-        if (discoveryShm_set(watcher->shmData, localNodePath, url) != CELIX_SUCCESS) {
-            celix_logHelper_log(discovery->loghelper, CELIX_LOG_LEVEL_WARNING, "Cannot set local discovery registration.");
-        }
-
-        discoveryShmWatcher_syncEndpoints(discovery);
-        sleep(5);
+    if (discoveryShm_set(watcher->shmData, localNodePath, url) != CELIX_SUCCESS) {
+        celix_logHelper_log(discovery->loghelper, CELIX_LOG_LEVEL_WARNING, "Cannot set local discovery registration.");
     }
 
-    return NULL;
+    discoveryShmWatcher_syncEndpoints(discovery);
```

Comment:
Network IO is involved?

Reply:
I do not think so, but will look into this